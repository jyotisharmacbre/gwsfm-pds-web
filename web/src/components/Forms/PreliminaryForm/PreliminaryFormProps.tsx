import EventType from "../../../enums/EventType";

export interface Props {
	onSave: (saveAll: boolean, event: EventType, prelimComponentDetails: any, index: number) => void;
	onPrevious: () => void;
	onToggle: (id: string) => void;
	preliminariesDetails: any;
	currencySymbol: string;
	isExpand: boolean;
	componentIdList: Array<string>;
	countryCode: string;
	insuranceRate: number;
	preliminaryState: any;
	subContractorState: any;
	discountState: any;
	projectStatus: number;
	event: EventType;
	loading: boolean;
}