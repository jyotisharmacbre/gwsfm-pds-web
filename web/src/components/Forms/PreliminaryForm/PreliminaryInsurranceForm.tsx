import React from 'react';
import { Field, formValueSelector } from 'redux-form';
import PdsFormInput from '../../PdsFormHandlers/PdsFormInput';
import { Validate, alphaNumeric, onlyNumber } from '../../../helpers/fieldValidations';
import { restrictMinus, restrictMinusAndDecimal} from '../../../helpers/utility-helper';
import {calculateSell, calculateTotalCost} from '../../../helpers/formulas';
import { IState } from '../../../store/state';
import { connect } from 'react-redux';
import { ICurrency } from '../../../store/Lookups/Types/ICurrency';
import { IPreliminariesComponentDetails } from '../../../store/Preliminaries/Types/IPreliminariesComponentDetails';
import { getPreliminarySummaryCalculation } from '../../../helpers/pricing-calculation-helper';
interface Props {
  fields:any;
      itemDetail:any;
      currencies:Array<ICurrency>|null;
      currencyId:number;
      currencySymbol:string;
      componentIndex:number;
      preliminaryData:Array<IPreliminariesComponentDetails>;
}  

const PreliminaryInsurranceForm:React.FC<Props>
= (props: Props) =>{
 
  const{fields,itemDetail,componentIndex,currencies,currencyId,currencySymbol,preliminaryData}=props;
  const updateCost=(index:number)=>
  {
    props.preliminaryData[props.componentIndex].items[index].totalCost=calculateTotalCost(getPreliminarySummaryCalculation(props.preliminaryData).cost);
  }
  return (
  <tbody>
  {props.fields.map((member, index) => (
   <tr key={index}>
     <td>
       <Field
                        name={`${member}.nameOfSupplier`}
                        data-test="nameOfSupplier"
                        type="text"
                        className="width-120 mb-0"
                        component={PdsFormInput}
                        validate={[
                            Validate.maxLength(1000),
                            alphaNumeric
                          ]}
                      />
    </td>
  <td>
    <Field
                name={`${member}.totalCost`}
                type="number"
                normalize={restrictMinus}
                input={{
                  value:calculateTotalCost(getPreliminarySummaryCalculation(props.preliminaryData).cost),
                    disabled: true,
                    onchange: updateCost(index)
                  
                  }}
                component={PdsFormInput}
                className="width-120 pl-20 required currency"
                validate={[
                    Validate.maxLength(15),
                    onlyNumber
                  
                ]}
                currency={currencySymbol}
                divPosition="relative"
              />
    </td>
    <td>
    <Field
                  name={`${member}.grossMargin`}
                  type="number"
                  normalize={restrictMinusAndDecimal}
                  component={PdsFormInput}
                  className="width-120 pl-20 required currency"
                  validate={[
                      Validate.maxLength(15),
                      onlyNumber
                    
                  ]}
                  currency={"%"}
                  divPosition="relative"
                />
    </td>
    <td>
    <Field
                  name={'totalSell'}
                  type="text"
                  input={{
                     value:calculateSell(props.preliminaryData[props.componentIndex].items[index].totalCost,props.preliminaryData[props.componentIndex].items[index].grossMargin),
                     disabled: true 
                    }}
                   component={PdsFormInput}
                  className="width-120 pl-20 required currency"
                  validate={[
                      Validate.maxLength(15),
                      onlyNumber
                  ]}
                  currency={currencySymbol}
                  divPosition="relative"
                />
    </td>
    <td>
    <Field
                        name={`${member}.comments`}
                        data-test="comments"
                        type="text"
                        component={PdsFormInput}
                        validate={[
                            Validate.maxLength(1000),
                            alphaNumeric
                          ]}
                      />
    </td>

     </tr>
    ))}
    </tbody>
  )};
  const mapStateToProps = (state: IState) => ({
    preliminaryData: selector(state,"preliminaryDetails")
   
  });
 const selector=formValueSelector("PreliminaryForm");
export default connect(mapStateToProps)(PreliminaryInsurranceForm);;

