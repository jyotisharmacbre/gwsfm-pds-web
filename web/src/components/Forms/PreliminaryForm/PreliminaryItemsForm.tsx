import React from 'react';
import { Field, formValueSelector } from 'redux-form';
import PdsFormInput from '../../PdsFormHandlers/PdsFormInput';
import { Validate, alphaNumeric, onlyNumber } from '../../../helpers/fieldValidations';
import { getFilterElementFromArray,calculateCost,calculateSell} from '../../../helpers/utility-helper';
import { IState } from '../../../store/state';
import { connect } from 'react-redux';
import { ICurrency } from '../../../store/Lookups/Types/ICurrency';
import { IPreliminariesComponentDetails } from '../../../store/Preliminaries/Types/IPreliminariesComponentDetails';
interface Props {
  fields:any;
      itemDetail:any;
      currencies:Array<ICurrency>|null;
      currencyId:number;
      currencySymbol:string;
      componentIndex:number;
      preliminaryData:Array<IPreliminariesComponentDetails>;
}  

const PreliminaryItemsForm:React.FC<Props>
= (props: Props) =>{
  const updateCost=(index:number)=>
  {
    props.preliminaryData[props.componentIndex].items[index].totalCost=(props.preliminaryData[props.componentIndex].items[index].noOfHours*props.preliminaryData[props.componentIndex].items[index].hourRate);
  }
  const{fields,itemDetail,componentIndex,currencies,currencyId,currencySymbol,preliminaryData}=props;
  const currencySymbols:string=getFilterElementFromArray(props.currencies,"currencyId",props.currencyId,"currencySymbol");

  return (
  <tbody>
  {props.fields.map((member, index) => (
   <tr>
     <td>{props.itemDetail.items[index].itemName}</td>
     <td>
                       {props.itemDetail.items[index].itemId=="2"||props.itemDetail.items[index].itemId=="3"?<Field
                        name={`${member}.nameOfSupplier`}
                        input={{disabled: true} }
                        data-test="nameOfSupplier"
                        type="text"
                        component={PdsFormInput}
                        validate={[
                            Validate.maxLength(1000),
                            alphaNumeric
                          ]}
                      />:<Field
                      name={`${member}.nameOfSupplier`}
                      data-test="nameOfSupplier"
                      type="text"
                      component={PdsFormInput}
                      validate={[
                          Validate.maxLength(1000),
                          alphaNumeric
                        ]}
                    />}
    </td>
    <td>
    {(props.itemDetail.items[index].itemId=="1"||props.itemDetail.items[index].itemId=="2")?<Field
                  name={`${member}.noOfHours`}
                  type="number"
                 input={{ disabled: true}}
                  component={PdsFormInput}
                  className="width-120 pl-20 required "
                  validate={[
                    Validate.maxLength(15),
                    onlyNumber
                  ]}
                  divPosition="relative"
                />:<Field
                name={`${member}.noOfHours`}
                type="number"
                component={PdsFormInput}
                className="width-120 pl-20 required "
                validate={[
                  Validate.maxLength(15),
                  onlyNumber
                ]}
                divPosition="relative"
              />}
    </td>
     <td>
                {(props.itemDetail.items[index].itemId=="1"||props.itemDetail.items[index].itemId=="2")?<Field
                  name={`${member}.hourRate`}
                  type="number"
                 input={{ disabled: true}}
                  component={PdsFormInput}
                  className="width-120 pl-20 required "
                  validate={[
                    Validate.maxLength(15),
                    onlyNumber
                  ]}
                  currency={currencySymbols}
                  divPosition="relative"
                  placeholderKey=""
                />:<Field
                name={`${member}.hourRate`}
                type="number"
                component={PdsFormInput}
                className="width-120 pl-20 required "
                validate={[
                  Validate.maxLength(15),
                  onlyNumber
                ]}
                currency={currencySymbols}
                divPosition="relative"
                placeholderKey=""
              />}
    </td>
  <td>
    {props.itemDetail.items[index].itemId!="3"&&props.itemDetail.items[index].itemId!="4"?<Field
                  name={`${member}.totalCost`}
                  type="number"
                  component={PdsFormInput}
                  className="width-120 pl-20 required"
                  validate={[
                    Validate.maxLength(15),
                    onlyNumber
                  ]}
                  currency={currencySymbols}
                  divPosition="relative"
                />:<Field
                name={`${member}.totalCost`}
                type="number"
                input={{
                  value:calculateCost(props.preliminaryData[props.componentIndex].items[index].noOfHours,props.preliminaryData[props.componentIndex].items[index].hourRate),
                  disabled: true,
                  onchange:updateCost(index)
                  }}
                component={PdsFormInput}
                className="width-120 pl-20 required"
                validate={[
                  Validate.maxLength(15),
                  onlyNumber
                ]}
                currency={currencySymbols}
                divPosition="relative"
              />}
    </td>
    <td>
    <Field
                  name={`${member}.grossMargin`}
                  type="number"
                  component={PdsFormInput}
                  className="width-120 pl-20 required"
                  validate={[
                    Validate.maxLength(15),
                    onlyNumber
                  ]}
                  currency={"%"}
                  divPosition="relative"
                />
    </td>
    <td>
    <Field
                  name={'totalSell'}
                  type="text"
                  input={{
                    value:calculateSell(props.preliminaryData[props.componentIndex].items[index].totalCost,props.preliminaryData[props.componentIndex].items[index].grossMargin),
                     disabled: true 
                    }}
                   component={PdsFormInput}
                  className="width-120 pl-20 required"
                  validate={[
                    Validate.maxLength(15),
                    onlyNumber
                  ]}
                  currency={currencySymbols}
                  divPosition="relative"
                />
    </td>
    <td>
    <Field
                        name={`${member}.comments`}
                        data-test="comments"
                        type="text"
                        component={PdsFormInput}
                        validate={[
                            Validate.maxLength(1000),
                            alphaNumeric
                          ]}
                      />
    </td>

     </tr>
    ))}
    </tbody>
  )};
  const mapStateToProps = (state: IState) => ({
    preliminaryData: selector(state,"preliminaryDetails"),
    currencies:state.lookup.currencies,
    currencyId:state.project.form.currencyId
  });
 const selector=formValueSelector("PreliminaryForm");
export default connect(mapStateToProps)(PreliminaryItemsForm);;

