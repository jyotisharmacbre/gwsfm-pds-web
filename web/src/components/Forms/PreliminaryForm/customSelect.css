
/*
*custom select option list
*/
.select-wrapper select {
    display: none; /*hide original SELECT element: */
  }
  
  /* .select-selected {
    background-color: DodgerBlue;
  } */
  
  /* Style the arrow inside the select element: */
  .select-selected:after {
    position: absolute;
    content: "";
    top: 14px;
    right: 10px;
    width: 0;
    height: 0;
    border: 6px solid transparent;
    border-color: #fff transparent transparent transparent;
  }
  
  /* Point the arrow upwards when the select box is open (active): */
  .select-selected.select-arrow-active:after {
    border-color: transparent transparent #fff transparent;
    top: 7px;
  }
  
  /* style the items (options), including the selected item: */
  .select-items div,.select-selected {
    cursor: pointer;
    max-width: 285px;
    font-size: 14px;
    color: #495057;
    background: #fff;
    -webkit-appearance: none;
    -moz-appearance: none;
    appearance: none;
    border: 1px solid #eaeef5;
    position: relative;
    border-radius: 6px;
    -moz-border-radius: 6px;
    -webkit-border-radius: 6px;
    padding: 10px 10px;
    height: auto !important;
  }
  
  /* Style items (options): */
  .select-items {
    position: absolute;
    background-color: DodgerBlue;
    top: 100%;
    left: 0;
    right: 0;
    z-index: 99;
  }
  
  /* Hide the items when the select box is closed: */
  .select-hide {
    display: none;
  }
  
  .select-items div:hover, .same-as-selected {
    background-color: rgba(0, 0, 0, 0.1);
  }