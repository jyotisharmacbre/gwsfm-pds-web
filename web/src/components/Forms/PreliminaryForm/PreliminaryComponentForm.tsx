import React from 'react';
// import { Col } from 'react-bootstrap';
import { Field, reduxForm, InjectedFormProps } from 'redux-form';
import { connect } from 'react-redux';
import PdsFormInput from '../../PdsFormHandlers/PdsFormInput';
import PdsFormSelect from '../../PdsFormHandlers/PdsFormSelect';
import PdsFormRadio from '../../PdsFormHandlers/PdsFormRadio';
import PdsFormTextArea from '../../PdsFormHandlers/PdsFormTextArea';
import PdsFormButton from '../../PdsFormHandlers/PdsFormButton';
import DatePicker from '../../DatePicker';
import { selectionButtons } from '../../../helpers/constants';
import { enquiryTypeData } from '../../../helpers/dropDownFormValues';
import { IState } from '../../../store/state';
import { IProjectAdditionalDetail } from '../../../store/ProjectOverviewForm/Types/IProjectAdditionalDetail';
import { IPreliminariesComponentDetails } from '../../../store/Preliminaries/Types/IPreliminariesComponentDetails';
import { getPropertyName, getDropdown } from '../../../helpers/utility-helper';
import { LookupType } from '../../../store/Lookups/Types/LookupType';
import EventType from '../../../enums/EventType';
import {
  projectStatusData,
  engagementData
} from '../../../helpers/dropDownFormValues';
import {
  alphaNumeric,
  onlyNumber,
  Validate
} from '../../../helpers/fieldValidations';
import { FormattedMessage } from 'react-intl';
import { IPreliminariesItemDetails } from '../../../store/Preliminaries/Types/IPreliminariesItemDetails';
import PreliminaryItemsForm from "./PreliminaryItemsForm"
interface Props {
    initialValues:IPreliminariesComponentDetails;
    componentName:string,
    componentId:string,
    isVisible:boolean
 }

let PreliminaryComponentForm: React.FC<
  Props & InjectedFormProps<IPreliminariesComponentDetails, Props>
> = (props: any) => {
  const handleCollapseEvent = (id: string) => {
    var element: any = document.getElementById('collapse_' + id);
    var isClassExists = element.classList.contains('show');
    if (isClassExists) {
      element.classList.add('hide');
      element.classList.remove('show');
    } else {
      element.classList.remove('hide');
      element.classList.add('show');
    }
  };
  return (
    <form>
            <div id="accordion">
      
          <div className="card">
            <div
              className="card-header"
              data-toggle="collapse"
              onClick={() => handleCollapseEvent(props.componentId)}
            >
              <a className="card-link" >{props.componentName}</a>
              <span className="fas fa-angle-up" aria-hidden="true"></span>
            </div>
            <div
              id={'collapse_' + props.componentId}
              className="hide expandAll"
              data-parent="#accordion"
            >
              
              {props.initialValues.items.map(function(itemData, itemKey) {
        return (
          <PreliminaryItemsForm 
          key={itemKey}
          form={`preliminaryItem_${itemKey}`}
          initialValues={itemData}>

          </PreliminaryItemsForm>
        );
      })}
        <div className="text-right" >
                  <button type="submit" className="text-right btn-sm">
                    SAVE
                  </button>
                </div>
             
            </div>
          </div>
        
      
    </div>
    </form>
  );
};


const form = reduxForm<IPreliminariesComponentDetails, Props>({
  destroyOnUnmount: false,
  forceUnregisterOnUnmount: false,
  enableReinitialize: true
})(PreliminaryComponentForm);

export default form;
