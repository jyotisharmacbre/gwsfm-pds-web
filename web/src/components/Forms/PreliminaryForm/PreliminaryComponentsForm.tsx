import React from 'react';
import { FieldArray} from 'redux-form';
import { FormattedMessage } from 'react-intl';
import PreliminaryItemsForm from './PreliminaryItemsForm';
import EventType from '../../../enums/EventType';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faAngleUp } from '@fortawesome/free-solid-svg-icons';
import PreliminaryInsurranceForm from './PreliminaryInsurranceForm';
import { CheckConstraints } from '../../../helpers/fieldValidations';

interface Props {
  submitHandler: (
        saveAll:boolean,
        event:EventType,
        prelimComponentDetails: any,
        index:number
      ) => void;
      onToggleEvent: (
        id: string
      ) => void;
      prelimData:any;
      handleSubmit:any;
      currencySymbol:string;
      
}  

const PreliminaryComponentsForm = ({ fields,submitHandler,handleSubmit,onToggleEvent,prelimData,currencySymbol}) => (

  <div>
  {fields.map((member, index) => (
    <div className="accordion" key={index}>
          <div className="card">
            <div
              className="card-header"
              data-toggle="collapse"
              
              
            >
              <a className="card-link" >{prelimData[index].componentName}</a>
             
              {!CheckConstraints(prelimData[index].componentId)
              ?<span aria-hidden="true">
              <FontAwesomeIcon data-test="collapse" className="active" icon={faAngleUp} onClick={()=>onToggleEvent(prelimData[index].componentId)}/>
              </span>:null}
            </div>
            <div
               id={"collapse_"+prelimData[index].componentId}
              className={CheckConstraints(prelimData[index].componentId)?"show expandAll":"hide expandAll"}
              data-test="toggle"
              data-parent="#accordion"
            > <form
            className="preliminary-form-1"
            onSubmit={handleSubmit}
            noValidate={true}
          >
                <div className="card-body pt-0">
            <div className="table-responsive price-sumry mt-0">
              <table className="table table-bordered">
                <thead>
                  <tr>
                    {!CheckConstraints(prelimData[index].componentId)?<th><FormattedMessage id="T_HEADING_ITEM" /></th>:null}
                    <th><FormattedMessage id="T_HEADING_NAME_OF_SUPPLIER" /></th>
                    {!CheckConstraints(prelimData[index].componentId)?<th><FormattedMessage id="T_HEADING_NO_OF_HOURS" /></th>:null}
                    {!CheckConstraints(prelimData[index].componentId)?<th><FormattedMessage id="T_HEADING_HOUR_RATE" /></th>:null}
                    <th><FormattedMessage id="T_HEADING_TOTAL_COST" /></th>
                    <th><FormattedMessage id="T_HEADING_GROSS_MARGIN" /></th>
                    <th><FormattedMessage id="T_HEADING_TOTAL_SELL" /></th>
                    <th><FormattedMessage id="T_HEADING_COMMENTS" /></th>
                  </tr>
                </thead>
                
                {!CheckConstraints(prelimData[index].componentId)?<FieldArray 
              name={`${member}.items`} 
              component={PreliminaryItemsForm}
              itemDetail={prelimData[index]}
              componentIndex={index}
              currencies={[]}
              currencyId={0}
              currencySymbol={currencySymbol}
              key={index}
            />:<FieldArray 
            name={`${member}.items`} 
            component={PreliminaryInsurranceForm}
            itemDetail={prelimData[index]}
            componentIndex={index}
            currencies={[]}
            currencyId={0}
            currencySymbol={currencySymbol}
            key={index}/>}
           
            </table>
                </div>
                </div>
                </form>
        <div className="text-right preliminary_btn" >
                  <button type="button" data-test="componentSave" className="text-center btn-sm" onClick={handleSubmit(values=>submitHandler(false,EventType.none,values,index))} >
                  <FormattedMessage id="BUTTON_SAVE" />
                  </button>
                </div>
             
            </div>
          </div>
     
           
        </div>
  
    ))}
    </div>
  );
 
export default PreliminaryComponentsForm;

