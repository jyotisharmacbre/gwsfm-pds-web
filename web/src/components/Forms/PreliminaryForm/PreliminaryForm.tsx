import React from 'react';
import { reduxForm,FieldArray, InjectedFormProps } from 'redux-form';
import { connect } from 'react-redux';
import { IState } from '../../../store/state';
import { FormattedMessage,injectIntl } from 'react-intl';
import { IPreliminaryForm } from '../../../store/Preliminaries/Types/IPreliminaryState';
import PreliminaryComponentsForm from './PreliminaryComponentsForm';
import IReactIntl from '../../../Translations/IReactIntl';
import EventType from '../../../enums/EventType';

interface Props {
  onSave: (saveAll:boolean,event:EventType,prelimComponentDetails: any,index:number) => void;
  onPrevious:()=>void;
  onToggle:(id:string)=>void;
  preliminariesDetails: any;
  currencySymbol:string;
  isExpand: boolean;
  componentIdList:Array<string>;
}
let PreliminaryForm: React.FC<
  Props &IReactIntl& InjectedFormProps<IPreliminaryForm, Props>
> = (props: any) => {
  const {handleSubmit} = props;
  return (
    <form
            className="preliminary-form"
            data-test="preliminary-form"
            onSubmit={handleSubmit}
            noValidate={true}
          >
             <FieldArray 
      name="preliminaryDetails" 
      component={PreliminaryComponentsForm}
      submitHandler={props.onSave}
      onToggleEvent={props.onToggle}
      prelimData={props.preliminariesDetails}
      handleSubmit={handleSubmit}
      currencySymbol={props.currencySymbol}
      isExpand= {props.isExpand}
      componentIdList={props.componentIdList}
       />
     <div className="mr-35 three-btn">
         <button
          className="active" 
          data-test="previous"
                type="button"
                name="previous"
                onClick={handleSubmit(values => props.onPrevious())}
              >
                <FormattedMessage id="BUTTON_PREVIOUS" />
              </button>
              <button
              name="save" className="active ml-auto" data-test="save"
                type="button"
                onClick={handleSubmit(values => props.onSave(true,EventType.none,values))}
              >
                <FormattedMessage id="BUTTON_SAVE" />
              </button>
              <button
                type="button"
                name="next"
                data-test="next"
                onClick={handleSubmit(values => props.onSave(true,EventType.next,values))}
              >
                <FormattedMessage id="BUTTON_NEXT" />
              </button>
      </div>
    </form>
  );
};

const mapStateToProps = (state: IState) => ({
  initialValues: state.preliminary,
});

const form = reduxForm<IPreliminaryForm, Props>({
  destroyOnUnmount: false,
  forceUnregisterOnUnmount: false,
  form: 'PreliminaryForm',
  enableReinitialize: true,
  keepDirtyOnReinitialize :true
})(injectIntl(PreliminaryForm));

export default connect(mapStateToProps)(form);
