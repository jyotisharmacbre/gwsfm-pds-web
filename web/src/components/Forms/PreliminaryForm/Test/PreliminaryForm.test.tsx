import React from 'react';
import { mount} from 'enzyme';
import { Provider } from 'react-redux';
import {findByTestAtrr} from '../../../../helpers/test-helper';
import { IntlProvider } from 'react-intl';
import translations from '../../../../Translations/translation';
import configureStore from 'redux-mock-store';
import { reducer as formReducer } from 'redux-form';
import PreliminaryForm from '../PreliminaryForm';
import {initialState} from "../../../../store/Preliminaries/InitialState"
import { preliminaryComponentIdList } from './PreliminaryFormTestData';

describe('Preliminary Form testCases', () => {
 
  const mockStore = configureStore([]);
  let store;
  let wrapper;
  const props: any = {
    onSave : jest.fn(),
    onToggle: jest.fn(),
    onPrevious:jest.fn(),
    preliminariesDetails:initialState,
    componentIdList:preliminaryComponentIdList

  }; 
  beforeEach(() => {
    store = mockStore({
      form: formReducer,
      preliminaryDetails: initialState
    });
    wrapper = mount(
      <Provider store={store}>
      <IntlProvider locale="en" messages={translations['en'].messages}>
      <PreliminaryForm {...props}/>
      </IntlProvider>
      </Provider>
    );
  });

  it('Defines the component', () => {
    expect(wrapper).toBeDefined();
  }); 
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('test previous button click event', () => {
    const button = findByTestAtrr(wrapper,'previous').first();
    button.simulate('click');
    expect(props.onPrevious.mock.calls.length).toEqual(1);
  });
  it('test next button click event', () => {
    const button = findByTestAtrr(wrapper,'next').first();
    button.simulate('click');
    expect(props.onSave.mock.calls.length).toEqual(1);
  });
  it('test save button click event', () => {
    const button = findByTestAtrr(wrapper,'save').first();
    button.simulate('click');
    expect(props.onSave.mock.calls.length).toEqual(2);
  });
 
  });
