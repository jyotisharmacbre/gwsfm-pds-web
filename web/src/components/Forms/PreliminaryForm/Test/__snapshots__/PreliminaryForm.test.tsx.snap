// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preliminary Form testCases should match the snapshot 1`] = `ReactWrapper {}`;

exports[`Sub Contractor Activity Form tests should match the snapshot 1`] = `ReactWrapper {}`;
